client.go

```diff
@@ -689,86 +839,167 @@ func (c *Client) SetRedirectPolicy(policies ...interface{}) *Client {
 		}
 		return nil // looks good, go ahead
 	}
-
+	c.lock.Unlock()
```

Comment:
@tttturtle-russ Any reason you didn't use `defer c.lock.Unlock()` next to lock line on 842?